README.md

```diff
@@ -1,289 +1,613 @@
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference`-able declaration to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference`-able declaration to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into the application.
+
+A Service Binding resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` must be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTranstionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A Service Binding resource **MAY** define `reason` and `message` entries to describe the last `status` transition.
+
+## Resource Type Schema
+
+```yaml
 apiVersion: service.binding/v1alpha1
 kind: ServiceBinding
 metadata:
-  name: example-service-binding
+  name:                 # string
 spec:
-  mountPathPrefix: "/var/bindings"
-  services:
-    - group: postgres.dev
-      kind: Service
-      name: global-user-db
-      version: v1beta1
-      id: postgres-global-user
-    - group: ibmcloud.ibm.com
-      version: v1alpha1
-      kind: Binding
-      name: coligo-service-binding
-      id: coligo-service-binding
-  application:
-    name: nodejs-rest-http-crud
-    group: apps
-    version: v1
-    resource: deployments
+  name:                 # string, optional, default: .metadata.name
+  type:                 # string, optional
+  provider:             # string, optional
+
+  application:          # PodSpec-able resource ObjectReference-able
+    apiVersion:         # string
+    kind:               # string
+    name:               # string
+    containers:         # []intstr.IntOrString, optional
+    ...
+
+  service:              # Provisioned Service-able resource ObjectReference-able
+    apiVersion:         # string
+    kind:               # string
+    name:               # string
+    ...
+
+status:
+  conditions:           # []Condition containing at least one entry for `Ready`
+  - type:               # string
+    status:             # string
+    lastTransitionTime: # Time
+    reason:             # string
+    message:            # string
 ```
 
-#### Customizing data bindings
+## Example Resource
 
-The `ServiceBinding` CR has an element per service called `dataMappings`, whereby the author of the CR can rename certain binding items and/or compose of multiple items:
+```yaml
+apiVersion: service.binding/v1alpha1
```

Comment:
[ echoing thoughts from the meeting ]

I feel we should not have the CRD schema as part of the specification.

The specification should have:
* How backing services are to be bindable.
* How applications could expect to consume them.

Any service binding implementation that is able to read from compliant backing services and project into apps written in a consumable manner.

We might want to 'suggest' that the request to create a Service Binding should/may include
- type
- provider
- one or more workloads
- one or more backing services
- one or more environment var mappings

If the specification specifies the minimal CRD, then we are basically saying that any change in the CRD/API needs a change in the spec as well.

Reply:
To me it feels like if the spec doesn't have a standard way for defining a link between an app and a service it will be a glaring omission.   Then it would be just a matter of time before another spec comes along and says "here's a standard way to declare your bindings!". 😄 

I think the `Ingress` example is a good one in terms of demonstrating a few things:
* a standard way to expose services outside of the cluster
* can be extended via annotations for framework specific behaviour (e.g. NGINX annotations)
* can be accepted as input and transformed into something more advanced (Ingress -> Route transformation in OCP)

I feel it would be good to take a similar path here, where frameworks can start by adding annotations to the `ServiceBinding` CR.  For example, let's say we have a framework that wants to be able to have a copy of all of the mounted items as env vars:

```
apiVersion: service.binding/v1alpha1
kind: ServiceBinding
metadata:
  name: online-banking-to-account-service
  annotations:
    binding-env-all: "true"
 ...
```

So at least there's a bare minimum bar that is interoperable, with value-add extensions.

Reply:
> Then it would be just a matter of time before another spec comes along and says "here's a standard way to declare your bindings!". 😄

Why is that an issue as long as the implementation understands the duck typing / annotations / descriptors which we want service providers to adhere to, and also projects things in an agreed upon way.

If enterprise A wants to deploy implementation X, what they deploy is a { CRD & Controller } combo. If they move to implementation Y, they deploy another { CRD & controller } combo.

If the Service Binding API becomes a part of Kubernetes, I'll be happy to accept the argument on one CRD & multiple controllers, otherwise breaking the API between alpha/beta/GA means breaking the specification - which is Okay too, but a cumbersome process :) 

( I don't intend to hold up the PR, we can move on :) )

Reply:
cool 😄  - created this one: https://github.com/application-stacks/service-binding-specification/issues/52